openc3-cosmos-script-runner-api/app/models/running_script.rb

```diff
@@ -1066,12 +1067,16 @@ def mark_breakpoint
   def run_text(text,
                 line_offset = 0,
                 text_binding = nil,
-                close_on_complete = false)
+                close_on_complete = false,
+                initial_filename: nil)
     initialize_variables()
     @line_offset = line_offset
     saved_instance = @@instance
     saved_run_thread = @@run_thread
-    @@instance   = self
+    @@instance = self
+    if initial_filename
+      OpenC3::Store.publish(["script-api", "running-script-channel:#{@id}"].compact.join(":"), JSON.generate({ type: :file, filename: initial_filename, text: text.to_utf8, breakpoints: [] }))
+    end
```

Comment:
`run_text` appears to only be called by run. I'm not sure who calls run but the file should already be published by `load_file_into_script` or `start` right? Why this new publish here?